fhf-datafangst-client/src/components/BenchmarkData/BenchmarkCards.tsx

```diff
@@ -0,0 +1,232 @@
+    dates.push(trip.start);
+  });
+  return dates;
+};
+
+enum BenchmarkType {
+  totalTime,
+  fishingHours,
+  fishingDistance,
+  fishingWeight,
+}
+
+export const BenchmarkCards: FC = () => {
+  const dispatch = useAppDispatch();
+  const trips = useAppSelector(selectTrips);
+  const numHistoric = useAppSelector(selectBenchmarkNumHistoric);
+  if (!trips) {
+    return <></>;
+  }
+
+  const totalTimes = getTotalTimes(trips);
+  const fishingHours = getFishingHours(trips);
+  const fishingDistance = getFishingDistance(trips);
+  const fishingWeight = getFishingWeight(trips);
+  const totalTimeMean =
+    totalTimes.reduce((a, b) => a + b, 0) / totalTimes.length;
+  const fishingHoursMean =
+    fishingHours.reduce((a, b) => a + b, 0) / fishingHours.length;
+  const fishingDistanceMean =
+    fishingDistance.reduce((a, b) => a + b, 0) / fishingDistance.length;
+  const fishingWeightMean =
+    fishingWeight.reduce((a, b) => a + b, 0) / fishingWeight.length;
+
+  const handleClick = (type: BenchmarkType) => {
+    const benchmarkModal: BenchmarkModalParams = {};
+    let data: number[];
+    let metric: string;
+    if (type === BenchmarkType.totalTime) {
+      benchmarkModal.title = "Total tid";
+      benchmarkModal.description =
+        "Total tid er regnet som tiden mellom havneavgang og havneanløp.";
+      metric = "Timer";
+      data = totalTimes;
+    } else if (type === BenchmarkType.fishingHours) {
+      benchmarkModal.title = "Fiske tid";
+      benchmarkModal.description =
+        "Fiske tid er regnet som summen av tiden brukt under hver fangstmelding.";
+      metric = "Timer";
+      data = fishingHours;
+    } else if (type === BenchmarkType.fishingDistance) {
+      benchmarkModal.title = "Fiske distanse";
+      benchmarkModal.description =
+        "Fiske distanse er regnet ut basert på vms/ais meldingene som ble sendt under hver fangstmelding";
+      metric = "Meter";
+      data = fishingDistance;
+    } else if (type === BenchmarkType.fishingWeight) {
+      benchmarkModal.title = "Total vekt";
+      benchmarkModal.description = "Total vekt er basert på total landet vekt";
+      metric = "Kilo";
+      data = fishingWeight;
+    } else {
+      return;
+    }
+    dispatch(setBenchmarkHistoric([metric, getTripDates(trips), data]));
+    dispatch(setBenchmarkModal(benchmarkModal));
+  };
+
+  return (
+    <Grid
+      container
+      spacing={3}
+      sx={{ marginTop: "3vh", backgroundColor: "primary.main" }}
+    >
+      <Grid item xs={6}>
+        <Box>
+          <BenchmarkCard
+            title="Total tid"
+            avatar={<AccessTimeIcon />}
+            value={totalTimes[0]}
+            description="Siste tur"
+            primary_color={
+              totalTimes[0] > totalTimeMean ? "#6CE16A" : "#93032E"
+            }
+            secondary_value={totalTimeMean}
+            secondary_description={
+              "Gjennomsnitt siste " + numHistoric.toString() + " turer"
+            }
```

Comment:
1 is not an issue, that will just give the same data as the previous trip. 0 would cause a division by zero error, but I would say that is a developer error if that becomes zero.

Reply:
I don't see how thats ok.
Currently, if I use an account that has zero trips, when I go to this page I get a white screen and a bunch of errors in the console.
This needs to be handled (i.e. telling the user they have no trips to show stats for)

Reply:
Right, of course. I think you shouldnt be allowed to see this page if you have zero trips. Ill try to fix something.